# Generated by Django 4.2.8 on 2024-02-02 13:13

from django.db import migrations


def edit_employee_in_sync_report(apps, _):
    """
    Меняем привязку пользователей в информации о сменах
    Employee -> CustomUser
    """
    CustomUser = apps.get_model('src', 'CustomUser')
    SyncReport = apps.get_model('src', 'SyncReport')

    for sync_report in SyncReport.objects.iterator():
        try:
            sync_report_employee = CustomUser.objects.get(
                name=sync_report.employee.name,
            )
            sync_report.new_employee = sync_report_employee
            sync_report.save()
        except CustomUser.DoesNotExist:
            print('Синхронизация. Проблема с записью id:', sync_report.id)
        except AttributeError:
            print('Синхронизация(a). Проблема с записью id:', sync_report.id)


class Migration(migrations.Migration):

    dependencies = [
        ('src', '0046_syncreport_new_employee'),
    ]

    operations = [
        migrations.RunPython(edit_employee_in_sync_report),
    ]
